Eugene/MovieSearch/my-app/src/components/MainPage/search-panel/SearchPanel.tsx

```diff
@@ -0,0 +1,117 @@
+import { parse, stringify } from "querystring";
+import React from "react";
+import "./SearchPanel.css";
+import { Location } from "history";
+
+interface ISearchPanelProps {
+  handleSearch(input: string, searchBy: string): void;
+  location: Location;
+}
+
+interface ISearchPanelState {
+  input: string;
+  searchBy: SearchBy;
+}
+
+enum SearchBy {
+  Title = "title",
+  Genre = "genres",
+}
+
+class SearchPanel extends React.Component<
+  ISearchPanelProps,
+  ISearchPanelState
+> {
+  state: ISearchPanelState = {
+    input: "",
+    searchBy: SearchBy.Title,
+  };
+
+  componentDidMount() {
+    const searchParams = this.props.location.search.slice(1);
+    const parsed = parse(searchParams) as {
+      search: string;
+      searchBy: string;
+    };
+
+    if (parsed.search) {
+      this.setState({ input: parsed.search });
+    } else {
+      this.setState({ input: "" });
+    }
+
+    if (parsed.searchBy) {
+      if (parsed.searchBy == "title") {
+        this.setState({ searchBy: SearchBy.Title });
+      } else {
+        this.setState({ searchBy: SearchBy.Genre });
+      }
+    } else {
+      this.setState({ searchBy: SearchBy.Title });
+    }
+  }
```

Comment:
You can make it simplier. SearchBy.Title is a default filter so you can just check 
```
if (parsed.searchBy && parsed.searchBy === SearchBy.Genre) {
          this.setState({ searchBy: SearchBy.Genre });
} else {
         this.setState({ searchBy: SearchBy.Title });
} 
```